wordpress-mobile/WordPress-iOS
# Makes Save As Draft non-destructive

Fixes #6761  

This makes the Save as Draft button in the options menu non-destructive (blue instead of red). It also fixed everything else that went wrong when no longer dismissing the window when it's tapped. 

* Now when the post is published/pushed to the server the post is replaced with a new revision if the window isn't immediately dismissed (matches original hybrid VC logic). 
* Made window dismissal logic more explicit instead of saying the secondary publish button was tapped. Code is much clearer to follow.

## To test:
1. Create a new post.
2. Tap the options menu. 
3. Verify Save as Draft is present and in non-destructive colors.
4. Tap Save as Draft.
5. Verify Publish button changes to Update and the window is not dismissed. Note: "dirty" logic isn't available yet so the publish button will continue to be enabled instead of disabling until more changes are made.

Next:

1. Edit an existing draft.
2. Tap the options menu.
3. Verify Publish Immediately is present and is in destructive colors (red).
4. Tap Publish Immediately.
5. Verify window dismisses after publishing is complete. (post-post will appear)

Needs review: @diegoreymendez, @rachelmcr (if you have time to install the alpha from Buddybuild)
